<h1 style="color:green;">MOORA (Multi-Objective Optimization by Ratio Analysis)</h1>

<p style="font-size:15px; font-weight:bold;"> MOORA (Multi-Objective Optimization by Ratio Analysis), çok kriterli karar verme problemlerinde (MCDM) alternatifler arasından en uygun olanı seçmek için kullanılan bir yöntemdir.</p>
<p style="font-size:15px; font-weight:bold;">Bu yöntem, her bir kriter için alternatiflerin performanslarını karşılaştırarak ve belirli bir oranlama sistemi kullanarak bir puanlama tablosu oluşturur. En yüksek puanı alan alternatif en uygun alternatif olarak seçilir.  </p>

<h3 style="color:green;">MOORA Yönteminin Diğer Yöntemlere Göre Farkları:</h3>

<h3 style="color:green;">1)TOPSIS</h3>
<p style="font-size:15px; font-weight:bold;"> - MOORA, ideal çözüme olan uzaklık ve referans noktasına olan yakınlık gibi kavramları kullanmaz.</p>
<p style="font-size:15px; font-weight:bold;"> - TOPSIS'te ideal ve anti-ideal çözümler önceden belirlenirken, MOORA'da referans noktası normalleştirilmiş performansların ortalaması olarak hesaplanır.</p>

<h3 style="color:green;">2)VIKOR </h3>
<p style="font-size:15px; font-weight:bold;"> - MOORA, VIKOR'da kullanılan "fazla arz" ve "fazla talep" fonksiyonlarını kullanmaz.</p>
<p style="font-size:15px; font-weight:bold;"> - VIKOR, uzlaşma ve karar verme endeksleri ile alternatifleri sıralarken, MOORA sadece puanlama tablosu kullanır.</p>

<h3 style="color:green;">3)PROMETHEE  </h3>
<p style="font-size:15px; font-weight:bold;"> - MOORA, PROMETHEE'de kullanılan tercihli eşitlikler ve tercihli dengesizlikler gibi kavramları kullanmaz.</p>
<p style="font-size:15px; font-weight:bold;"> -PROMETHEE, alternatifler arasındaki eşitlik ve dengesizlikleri değerlendirerek bir sıra oluştururken, MOORA her kriter için puanlama yapar.</p>

<h2 style="color:green;">MOORA Yöntemini Tercih Etme Durumları:

</h2>
<p style="font-size:15px; font-weight:bold;"> - Birden fazla kritere göre değerlendirilen alternatifler arasında en uygun olanı seçmek istiyorsanız.</p>
<p style="font-size:15px; font-weight:bold;"> - Farklı birimlere sahip kriterleri karşılaştırılabilir hale getirmek istiyorsanız.
</p>
<p style="font-size:15px; font-weight:bold;"> - Her bir kriterin önemini ve yönünü dikkate almak istiyorsanız.</p>
<p style="font-size:15px; font-weight:bold;"> - Basit ve anlaşılır bir yöntem kullanmak istiyorsanız.</p>

<h3 style="color:green;">MOORA Yönteminin Avantajları:</h3>
<p style="font-size:15px; font-weight:bold;"> - Basit ve anlaşılır bir yöntemdir.
</p>
<p style="font-size:15px; font-weight:bold;"> - Farklı birimlere sahip kriterleri karşılaştırılabilir hale getirmek için normalizasyon kullanır.

</p>
<p style="font-size:15px; font-weight:bold;"> - Her bir kriterin önemini ve yönünü dikkate alan bir oranlama sistemi kullanır.
</p>
<p style="font-size:15px; font-weight:bold;"> - Hesaplamaları kolaydır ve manuel olarak veya yazılım kullanılarak yapılabilir</p>


<h3 style="color:green;">MOORA Yönteminin Dezavantajları:</h3>
<h5 style="color:green;">Kriterler Arasındaki Etkileşimleri Dikkate Almaz:</h5>

<p style="font-size:15px; font-weight:bold;"> - MOORA yöntemi, kriterler arasındaki etkileşimleri veya sinerjileri dikkate almaz. Bu, bazı durumlarda sonuçların gerçek durumu tam olarak yansıtmamasına neden olabilir. Örneğin, bir eğitim kurumunda hem giriş sınav puanı hem de mülakat puanı önemli kriterler olabilir.
</p>
<p style="font-size:15px; font-weight:bold;"> - MOORA yöntemi her iki kritere de eşit ağırlık verirken, mülakat puanı yüksek olan bir adayın düşük giriş sınav puanına sahip olmasına rağmen daha yüksek bir puan almasına neden olabilir. Bu durumda, mülakat puanı ve giriş sınav puanı arasındaki etkileşim göz ardı edilmiş olur.
</p>

<h5 style="color:green;">Kriterlerin Ağırlıklarını Önceden Belirlemek Gerekir:</h5>

<p style="font-size:15px; font-weight:bold;"> - MOORA yönteminde her kriter için bir ağırlık değeri belirlemek gerekir. Bu ağırlıklar, kriterlerin önemini ve karar vermedeki rolünü temsil eder.
</p>
<p style="font-size:15px; font-weight:bold;"> - Ağırlıkların belirlenmesi öznel bir işlemdir ve farklı kişiler farklı ağırlıklar belirleyebilir. Bu durum, farklı kişilerin MOORA yöntemini uyguladıklarında farklı sonuçlar almasına neden olabilir.
</p>

<h1 style="color:green;">MOORA ADIMLARI</h1>


<h4 style="color:purple;">1) Kriterlerin Belirlenmesi:</h4>
<p style="font-size:15px; font-weight:bold;"> Karar verme probleminizdeki tüm kriterleri belirleyin. Her bir kriter için faydaya mı (maksimize) yoksa maliyete mi (minimize) odaklandığınızı tanımlayın. Örneğin, bir araba seçerken yakıt verimliliği maksimize edilirken, fiyat minimize edilir.
</p>


<h4 style="color:purple;">2) Normalizasyon </h4>
<p style="font-size:15px; font-weight:bold;">Farklı birimlere sahip kriterleri 0 ile 1 arasında bir aralığa dönüştürün. Bu, kriterlerin birbirleriyle karşılaştırılabilir hale gelmesini sağlar.
</p>


<h5 style="color:purple;">a) Fayda Kriterleri:</h5>

$$
x_{ij} = \frac{v_{ij} - \min_{k} v_{kj}}{\max_{k} v_{kj} - \min_{k} v_{kj}}
$$
<h5 style="color:purple;"> b) Maliyet Kriterleri:</h5>

$$
x_{ij} = \frac{\max_{k} v_{kj} - v_{ij}}{\max_{k} v_{kj} - \min_{k} v_{kj}}
$$

$ x_{ij}$: $i$. alternatifteki $j$. kriterin normalize edilmiş değeri.\

$v_{ij}$: $i$. alternatifteki $j$. kriter değeri.\

$v_{kj}$: $k$. kriterin tüm alternatiflerdeki değerleri.\

$\min_{k} v_{kj}$: $k$. kriterin minimum değeri.\

$\max_{k} v_{kj}$: $k$. kriterin maksimum değeri.

<h5 style="color:purple;">Örnek:</h5>

Diyelim ki bir probleminizde 3 alternatif (A, B, C) ve 2 kriter (K1, K2) var. A alternatifinin K1 kriterindeki değeri 0.5 ve K2 kriterindeki değeri 0.8 olsun. Ayrıca, K1 kriterinin minimum değeri 0.2 ve maksimum değeri 0.9, K2 kriterinin minimum değeri 0.6 ve maksimum değeri 1.0 olsun. Bu durumda, A alternatifinin K1 ve K2 kriterleri için normalize edilmiş değerleri şu şekilde hesaplanır:

<h5 style="color:purple;">K1:</h5>

$$
x_{A1} = \frac{0.9-0.2}{0.5-0.2} = 0.4286
$$

<h5 style="color:purple;">K2:</h5>

$$
x_{A2} = \frac{1.0-0.6}{1.0-0.8} = 0.2
$$



<h5 style="color:purple;">Örnek 2 :</h5>
Diyelim ki yukarıdaki örnekten A alternatifinin normalize edilmiş değerleri K1 için 0.4286 ve K2 için 0.2 olsun. Ayrıca, K1 kriterinin minimum normalize edilmiş değeri 0.2 ve K2 kriterinin minimum normalize edilmiş değeri 0.1 olsun. Bu durumda, A alternatifinin K1 ve K2 kriterleri için oran değerleri şu şekilde hesaplanır:

K1:
$$
r_{A1} = \frac{0.2}{0.4286} = 2.143
$$

K2:
$$
r_{A2} = \frac{0.1}{0.2} = 2.0
$$



<h4 style="color:purple;">3) Performans Tablosu Oluşturma: </h4>
<p style="font-size:15px; font-weight:bold;"> Her bir alternatif için her bir kriterdeki performans değerlerini bir tabloda gösterin. Bu tablo, karar verme probleminizdeki tüm alternatifleri ve kriterleri kapsamalıdır.
</p>


<h4 style="color:purple;">4 ) Oranlama Sistemleri:</h4>

<h5 style="color:purple;">4.1)Oran Oranlama Sistemi:</h5>
<p style="font-size:15px; font-weight:bold;"> Bu sistemde, her bir alternatif için kriter değeri, en kötü alternatifteki kriter değerine oranlanır. En düşük kriter değerine sahip alternatif için oran değeri 1 olurken, diğer alternatifler için oran değeri 1'den büyük olur.
</p>

$$
r_{ij} = \frac{x_{ij}}{\min_{k} x_{kj}}
$$

$r_{ij}$: $i$. alternatifteki $j$. kriterin oran değeri.\

$m$: alternatif sayısı.


<h5 style="color:purple;">4.2)Referans Noktasına Dayalı Oranlama Sistemi:</h5>
<p style="font-size:15px; font-weight:bold;"> Bu sistemde, her bir alternatif için kriter değeri, tüm alternatiflerin kriter değerlerinin ortalamasına (referans noktası) oranlanır. Ortalama kriter değerine sahip alternatif için oran değeri 1 olurken, diğer alternatifler için oran değeri 1'den büyük veya küçük olabilir.
</p>

$$
r_{ij} = \frac{x_{ij}}{\frac{1}{m} \sum_{k=1}^{m} x_{kj}}
$$

$r_{ij}$: $i$. alternatifteki $j$. kriterin oran değeri.\

$m$: alternatif sayısı.


<h5 style="color:purple;">Seçim:</h5>
<p style="font-size:15px; font-weight:bold;"> Hangi oranlama sisteminin kullanılacağı, kriterlerin türüne ve karar verme probleminin özelliklerine bağlıdır. Genel olarak, Oran Oranlama Sistemi, kriterlerin birbirleriyle benzer olduğu durumlarda daha uygun olabilirken, Referans Noktasına Dayalı Oranlama Sistemi, kriterlerin farklı birimlere sahip olduğu veya farklı ölçeklerde ölçüldüğü durumlarda daha uygun olabilir.
</p>



<h5 style="color:purple;">5)Puanlama Tablosu:</h5>
<p style="font-size:15px; font-weight:bold;"> Oranlama sisteminden elde edilen oran değerleri, her bir alternatif için puanlama tablosu oluşturmak için kullanılır. Puanlama tablosu, her bir alternatif için her bir kriterdeki puanı gösterir. Puanlar genellikle 0 ile 1 arasında bir aralıkta olur.
</p>

$$
S_i = \sum_{j=1}^{n} w_j r_{ij}
$$

$S_i$: $i$. alternatifin toplam puanı.

$w_j$: $j$. kriterin ağırlığı (opsiyonel).

$r_{ij}$: $i$. alternatifteki $j$. kriterin oran değeri.


<h4 style="color:purple;">6)Sıralama:</h4>
<p style="font-size:15px; font-weight:bold;">Puanlama tablosundaki toplam puanlara göre alternatifler sıralanır. En yüksek toplam puana sahip alternatif, en uygun alternatif olarak kabul edilir.
</p>

In [1]:
import numpy as np

alternatif_sayisi = 3
kriter_sayisi = 2

alternatif_performanslari = np.random.rand(alternatif_sayisi, kriter_sayisi)

kriter_performanslari = np.random.rand(kriter_sayisi, alternatif_sayisi)

print("Alternatif Performansları:")
print(alternatif_performanslari)
print("\nKriter Performansları:")
print(kriter_performanslari)

Alternatif Performansları:
[[0.6852755  0.73791055]
 [0.76415002 0.30326523]
 [0.98646639 0.32335538]]

Kriter Performansları:
[[0.96059922 0.72135521 0.87759426]
 [0.98919699 0.39721145 0.59796235]]


In [2]:
def normalize_fayda_kriterleri(v_ij, v_kj):
    return (v_ij - min(v_kj)) / (max(v_kj) - min(v_kj))

def normalize_maliyet_kriterleri(v_ij, v_kj):
    return (max(v_kj) - v_ij) / (max(v_kj) - min(v_kj))

# Örnek veriler
v_ij = [0.5, 0.8]
v_kj = [[0.2, 0.5, 0.8], [0.6, 0.8, 1.0]]
m = alternatif_sayisi  # alternatif sayısı
w_j = [0.3, 0.7]  # kriter ağırlıkları
# Normalizasyon
x_A1 = normalize_fayda_kriterleri(v_ij[0], v_kj[0])
x_A2 = normalize_fayda_kriterleri(v_ij[1], v_kj[1])

print("K1 için normalize edilmiş değer (A):", x_A1)
print("K2 için normalize edilmiş değer (A):", x_A2)


K1 için normalize edilmiş değer (A): 0.4999999999999999
K2 için normalize edilmiş değer (A): 0.5000000000000001


In [3]:
def oran_oranlama_sistemi(x_ij, x_kj):
    return x_ij / min(x_kj)

def referans_noktasına_dayalı_oranlama_sistemi(x_ij, x_kj, m):
    return x_ij / (sum(x_kj) / m)

# Oranlama Sistemleri
r_A1 = oran_oranlama_sistemi(x_A1, v_kj[0])
r_A2 = oran_oranlama_sistemi(x_A2, v_kj[1])

print("K1 için oran değeri (A):", r_A1)
print("K2 için oran değeri (A):", r_A2)

# Referans Noktasına Dayalı Oranlama Sistemi
r_A1_ref = referans_noktasına_dayalı_oranlama_sistemi(x_A1, v_kj[0], m)
r_A2_ref = referans_noktasına_dayalı_oranlama_sistemi(x_A2, v_kj[1], m)

print("\nK1 için referans noktasına dayalı oran değeri (A):", r_A1_ref)
print("K2 için referans noktasına dayalı oran değeri (A):", r_A2_ref)


K1 için oran değeri (A): 2.499999999999999
K2 için oran değeri (A): 0.8333333333333336

K1 için referans noktasına dayalı oran değeri (A): 0.9999999999999998
K2 için referans noktasına dayalı oran değeri (A): 0.6250000000000002


In [4]:
def oran_oranlama_sistemi(x_ij, x_kj):
    return x_ij / min(x_kj)

def referans_noktasına_dayalı_oranlama_sistemi(x_ij, x_kj, m):
    return x_ij / (sum(x_kj) / m)

# Oranlama Sistemleri
r_A1 = oran_oranlama_sistemi(x_A1, v_kj[0])
r_A2 = oran_oranlama_sistemi(x_A2, v_kj[1])

print("K1 için oran değeri (A):", r_A1)
print("K2 için oran değeri (A):", r_A2)

# Referans Noktasına Dayalı Oranlama Sistemi
r_A1_ref = referans_noktasına_dayalı_oranlama_sistemi(x_A1, v_kj[0], m)
r_A2_ref = referans_noktasına_dayalı_oranlama_sistemi(x_A2, v_kj[1], m)

print("K1 için referans noktasına dayalı oran değeri (A):", r_A1_ref)
print("K2 için referans noktasına dayalı oran değeri (A):", r_A2_ref)



K1 için oran değeri (A): 2.499999999999999
K2 için oran değeri (A): 0.8333333333333336
K1 için referans noktasına dayalı oran değeri (A): 0.9999999999999998
K2 için referans noktasına dayalı oran değeri (A): 0.6250000000000002


In [5]:
def puanlama(w_j, r_ij):
    return sum(w * r for w, r in zip(w_j, r_ij))

# Puanlama
S_A = puanlama(w_j, [r_A1, r_A2])

print("Alternatif A'nın toplam puanı:", S_A)



Alternatif A'nın toplam puanı: 1.333333333333333


<h4 style="color:purple;">YORUM </h4>
<p style="font-size:15px; font-weight:bold;">Bu verilerle MOORA yöntemini uyguladığımızda, alternatif A'nın toplam puanı 1.3333 olarak hesaplanmıştır. Bu puan, alternatifin kriterlerin ağırlıklarına göre değerlendirildiğinde ne kadar iyi performans gösterdiğini gösterir.</p>

<p style="font-size:15px; font-weight:bold;">K1 kriteri için normalize edilmiş değer 0.5 ve K2 kriteri için normalize edilmiş değer 0.5 olarak bulunmuştur. Bu, alternatifin her iki kriterde de orta seviyede performans gösterdiğini gösterir.</p>

<p style="font-size:15px; font-weight:bold;">K1 kriteri için oran değeri 2.5 ve K2 kriteri için oran değeri 0.8333 olarak hesaplanmıştır. Bu, alternatifin K1 kriterinde diğer alternatiflere kıyasla çok daha iyi performans gösterdiğini ancak K2 kriterinde daha düşük performans gösterdiğini gösterir.</p>

<p style="font-size:15px; font-weight:bold;">Referans noktasına dayalı oranlama sistemi sonuçlarına göre, alternatifin K1 kriterinde ortalamanın üzerinde, ancak K2 kriterinde ortalamanın altında performans gösterdiği görülmektedir. Bu da alternatifin genel olarak orta seviyede bir performans sergilediğini işaret etmektedir.</p>

<p style="font-size:15px; font-weight:bold;">Sonuç olarak, alternatif A'nın toplam puanı 1.3333'tür ve bu da alternatifin ortalama bir performans sergilediğini gösterir. K1 kriterinin ağırlığı daha yüksek olduğu için, alternatifin performansı özellikle K1 kriterine bağlıdır. Ancak, K2 kriterinde de düşük bir performans göstermektedir. Bu bilgi, alternatif A'nın güçlü ve zayıf yönlerini belirlemek için değerli bir rehber sağlar.</p>
